### PR TITLE
ODBC+TLS configuration

### PR DESCRIPTION
**Pull request by arkgil**
This PR adds documentation on how to configure ODBC connection with TLS.

### CODE REVIEW

File: doc/Advanced-configuration.md [hunk 1]
```diff
@@ -319,6 +319,31 @@ An example configuration can look as follows:
 `{odbc_server, {pgsql, "localhost", "username", "database", "pass",
                [{ssl, required}, {ssl_opts, [{verify, verify_peer}, {cacertfile, "path/to/cacert.pem"}]}]}}.`
 
+### ODBC SSL connection setup
+
+If you've configured MongooseIM to use ODBC driver, i.e. you've provided ODBC connection string to `odbc_server` option, e.g.
```
Comment (goddammit) [inline doc/Advanced-configuration.md]: If you've configured MongooseIM to use the ODBC driver, i.e. you've provided an ODBC connection string to `odbc_server` option, e.g.

File: doc/Advanced-configuration.md [hunk 2]
```diff
@@ -319,6 +319,31 @@ An example configuration can look as follows:
...
+
+```erlang
+{odbc_server, "DSN=mydb"}.
+```
+
+then SSL options, along other connection options, should be present in `~/.odbc.ini` file.
```
Comment (goddammit) [inline doc/Advanced-configuration.md]: then the SSL options, along other connection options, should be present in the `~/.odbc.ini` file.

File: doc/Advanced-configuration.md [hunk 3]
```diff
@@ -319,6 +319,31 @@ An example configuration can look as follows:
...
+
+To enable SSL connection `sslmode` option needs to be set to `verify-full`.
```
Comment (goddammit) [inline doc/Advanced-configuration.md]: To enable the SSL connection the `sslmode` option needs to be set to `verify-full`.

File: doc/Advanced-configuration.md [hunk 4]
```diff
@@ -319,6 +319,31 @@ An example configuration can look as follows:
...
+Additionally, you can provide path to the CA certificate using `sslrootcert` option.
```
Comment (goddammit) [inline doc/Advanced-configuration.md]: Additionally, you can provide the path to the CA certificate using the `sslrootcert` option.

**Comment by arkgil** [comment]
@goddammit done 🎉

File: doc/Advanced-configuration.md [hunk 5]
```diff
@@ -247,7 +247,7 @@ See [Database backends configuration](./advanced-configuration/database-backends
 
 * **odbc_server** (local)
     * **Description:** SQL DB connection configuration. Currently supported DB types are `mysql` and `pgsql`.
-    * **Syntax:** `{odbc_server, {Type, Host, Port, DBName, Username, Password}}.`
+    * **Syntax:** `{odbc_server, {Type, Host, Port, DBName, Username, Password}}.` **or** ODBC connection string.
```
Comment (kzemek) [inline doc/Advanced-configuration.md]: This reads to me like `ODBC connection string` *replaces* the whole `{odbc_server, ...}` tuple

File: doc/Advanced-configuration.md [hunk 6]
```diff
@@ -319,6 +319,31 @@ An example configuration can look as follows:
 `{odbc_server, {pgsql, "localhost", "username", "database", "pass",
                [{ssl, required}, {ssl_opts, [{verify, verify_peer}, {cacertfile, "path/to/cacert.pem"}]}]}}.`
 
+### ODBC SSL connection setup
+
+If you've configured MongooseIM to use an ODBC driver, i.e. you've provided an ODBC connection string to `odbc_server` option, e.g.
```
Comment (kzemek) [inline doc/Advanced-configuration.md]: `odbc` in `odbc_server` really needs to go, it's very confusing that normally you would *not* use ODBC driver but still configure things with `odbc_*` options.

File: doc/Advanced-configuration.md [hunk 7]
```diff
@@ -319,6 +319,31 @@ An example configuration can look as follows:
...
+```erlang
+{odbc_server, "DSN=mydb"}.
+```
+
+then the SSL options, along other connection options, should be present in the `~/.odbc.ini` file.
+
+To enable SSL connection the `sslmode` option needs to be set to `verify-full`.
+Additionally, you can provide the path to the CA certificate using the `sslrootcert` option.
+
+#### Example configuration
```
Comment (kzemek) [inline doc/Advanced-configuration.md]: I guess that's an example `~/.odbc.ini` configuration?

File: doc/Advanced-configuration.md (hunk 6 above)
Comment (arkgil) [reply doc/Advanced-configuration.md]: Good point, that's could be the subject of another PR. @fenek @michalwski

Comment (fenek) [reply doc/Advanced-configuration.md]: Definitely, it is something that has been annoying me for quite a long time.

**Pull request merged**